### PR TITLE
Transform plg_media-action-rotate/rotate.js to ES6

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue # https://github.com/joomla-projects/joomla-es6/issues/27.

### CODE REVIEW

File: media/plg_media-action_rotate/js/rotate.es6.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Joomla = window.Joomla || {};
+
+Joomla.MediaManager = Joomla.MediaManager || {};
+Joomla.MediaManager.Edit = Joomla.MediaManager.Edit || {};
+
+(function mediaActionRotate() {
+  'use strict';
+
+  // Update image
+  const rotate = function rotate(angle) {
```
Comment (C-Lodder) [inline media/plg_media-action_rotate/js/rotate.es6.js]: const rotate = (angle) => {

File: media/plg_media-action_rotate/js/rotate.es6.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    document.getElementById('jform_rotate_angle').value = angle;
+
+    // Update the height input box
+    document.getElementById('jform_rotate_a').value = angle;
+
+    // Notify the app that a change has been made
+    window.dispatchEvent(new Event('mediaManager.history.point'));
+  };
+
+  const initRotate = function initRotate() {
```
Comment (C-Lodder) [inline media/plg_media-action_rotate/js/rotate.es6.js]: `const initRotate = () => {`

**Comment by C-Lodder** [comment]
Missing arrow functions

File: media/plg_media-action_rotate/js/rotate.es6.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
+/**
+ * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+Joomla = window.Joomla || {};
+
+Joomla.MediaManager = Joomla.MediaManager || {};
+Joomla.MediaManager.Edit = Joomla.MediaManager.Edit || {};
+
+(function mediaActionRotate() {
```
Comment (dneukirchen) [inline media/plg_media-action_rotate/js/rotate.es6.js]: here we can also use arrow function 
```
(() => {
...
})()
```

Comment (astridx) [reply media/plg_media-action_rotate/js/rotate.es6.js]: I do not understand when to use arrow functions and when to use named functions. Wouldn't it be better to leave the name of the function? If we leave off the function name then when the function throws an exception we do not see a name in the stack trace.

**Review by dneukirchen — APPROVED** [review]

File: media/plg_media-action_rotate/js/rotate.es6.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    document.getElementById('jform_rotate_a').value = angle;
+
+    // Notify the app that a change has been made
+    window.dispatchEvent(new Event('mediaManager.history.point'));
+  };
+
+  const initRotate = () => {
+    const funct = () => {
+      // The listeners
+      document.getElementById('jform_rotate_angle').addEventListener('change', function jformRotateAngle() {
```
Comment (dneukirchen) [inline media/plg_media-action_rotate/js/rotate.es6.js]: we can use arrow functions here.

File: media/plg_media-action_rotate/js/rotate.es6.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        rotate(parseInt(this.value, 10));
+      });
+      document.getElementById('jform_rotate_a').addEventListener('input', function jformRotateA() {
```
Comment (dneukirchen) [inline media/plg_media-action_rotate/js/rotate.es6.js]: we can use arrow functions here.

**Comment by dneukirchen** [comment]
rest looks good. thank you @astridx

File: media/plg_media-action_rotate/js/rotate.es6.js (hunk 5 above)
Comment (astridx) [reply media/plg_media-action_rotate/js/rotate.es6.js]: I think it is not possible because of this, is it?

Comment (dneukirchen) [reply media/plg_media-action_rotate/js/rotate.es6.js]: it is... you can access the value of the input element like so:

```
...addEventListener('input', (event) => {
    console.log(event.currentTarget.value);
    console.log(event.target.value);
});
```

fyi: 

* event.target is the element that triggered the event (e.g., the user clicked on)
* event.currentTarget is the element that the event listener is attached to.

**Review by dneukirchen — APPROVED** [review]

**Pull request merged**